libs/gl-plugin/src/node/mod.rs

```diff
@@ -1127,10 +1129,11 @@ where
                 }
 
                 trace!(
-                    "Got a request for {} with pubkey={}, sig={} and body size={:?}",
+                    "Got a request for {} with pubkey={}, sig={}, rune={} and body size={:?}",
```

Comment:
Notice that for this to be safe (i.e., node can't steal the rune and use it for its own commands) we need to either commit to the request in the Rune (append the command as a constraint at the end of the rune) or have the run commit to the pubkey used to sign the request. The latter is more flexible as it doesn't require keeping the raw rune around on the client, which might leak accidentally, or require the serialized payload at rune-verification time.

Reply:
I think you'll likely have considered that in the followup commit by the way, I just wanted to make sure that I check it myself :-)

Reply:
Always good to double check on critical parts :+1: 

We already build the rune with the public key in the `id` field of the rune. This binds the rune to the public key. The rune is based on the `authcode` of the signing device, that itself is based on a derived key (hkd) from the signer. This should commit us to the public key as well as attest the public key.

But as you said, we could of course even go one step further and use _one-time-runes_ (otrs ^^) and commit the rune to the signed request by adding the signature to the rune.